### DISCUSSION TITLE
Hello guys...

### DISCUSSION THREAD

**Discussion by WebShells** [opensourcepos/opensourcepos#3659] [Team Posts]
Discussions about the project since we do not have each other direct contact info, might be helpful... etc...

**Comment by daN4cat** [comment]
Spotted this only today.... sorry.

**Comment by jekkos** [comment]
Ok very useful did not know about this feature

**Comment by WebShells** [comment]
Jekkos seems to be on vacation still? 🤔

**Comment by jekkos** [comment]
Hi no back since some days, but busy times at work. However soon will be off again for two weeks

**Comment by WebShells** [comment]
Guys, demo blank page on sales completion

**Comment by WebShells** [comment]
Welcome back @jekkos

**Comment by SteveIreland** [comment]
What's the general feature configuration that resulted in the blank page?

**Comment by WebShells** [comment]
I have no idea... But is it still the same ? Am retrying now... I see blank page as well... Can somebody check his side in order to confirm ? Maybe the problem is just appearing to me ? @SteveIreland

**Comment by odiea** [comment]
I tried yesterday and had no issue but this morning the receipt is not showing on Sale completion. Previous receipts show properly.  Does not show any error in ff inspector. Status 200 Method Post.

**Comment by WebShells** [comment]
Yes @odiea, old receipts from takings windows or reports show regular receipt without problems... But on sale completion its displaying blank page...

**Comment by odiea** [comment]
I tried an Invoice and also changed to short receipt and still have the same issue.

**Comment by odiea** [comment]
Suspending Sales is not working and I tried changing Localization the page shows it is changed but has no effect on the rest of the language display.

**Comment by WebShells** [comment]
I tried this too, stays in same language... Ill try to refresh the build...

**Comment by WebShells** [comment]
Works fine after a reset, no idea what caused the issue before.

**Comment by odiea** [comment]
I downloaded and installed the prerequisites to compile ospos in win 10. It appeared to do something but it did not create the Dist folder. Any Ideas. Composer,node.js and grunt were installed.

**Comment by odiea** [comment]
This is the Install help I used. I guess if I am going to test other peoples repositories I better learn how to do it.
https://github.com/opensourcepos/opensourcepos/wiki/Installing-%22opensourcepos%22-in-windows-and-localhost

**Comment by odiea** [comment]
I just had to restart windows and now it works properly

**Comment by WebShells** [comment]
Mmm not a fan of win10 :x

**Comment by daN4cat** [comment]
Next will be to learn how to run Docker :-)

I managed to get it to run on my local laptop, I found the issue that caused the problem and I'll commit a small change.
I mean if you want to just run the standard Docker using docker-composer, it was returning URLs with https and not http even if FORCE_HTTPS=false. The UI was without css and js due to https being not valid.

**Comment by odiea** [comment]
Here is the procedure I use. Is this correct or is there a better way.
composer install
    npm install --only=dev
    bower install
    grunt -f
    grunt cssmin
I have tried running this bin/devinstall.sh but it stalls on the install php and says file not found.

**Comment by jekkos** [comment]
Ok maybe I should add some of the tips and tricks I often use with docker and my Dev setup. Once you start changing branches it's best to clear out existing containers to make sure it recreates the MySQL volume for instance.

**Comment by daN4cat** [comment]
Interesting stuff from our friend the daily troller: https://github.com/osposweb/osposweb/tree/osposweb-osposweb
http://osposweb.com/cliente/index.php/Systems

shooting to promote his business....

can we please ban him from this GitHub space? There is an option to block him completely, he shows no respect.

**Comment by daN4cat** [comment]
He writes:

..a fork from original OSPOS at https://github.com/opensourcepos/opensourcepos,which has improvements respect users that are filling a "bad actitude" from official project,

2. Better user support, no bad actitude!

**Comment by SteveIreland** [comment]
It is funny though, a guy with the worst open source attitude hyping himself as better than everyone else.

I personally think that anyone who deals with him deserves what they get.

Isn't he violating the license?  Does GitHub do anything to insure that participants abide by licenses?  Just wondering. 

I'm not sure it would be a good idea to ban him.  With his attitude I think he'd probably do what he could to cause some damage, probably coming back with a fake profile.  At least now we can call him out on bad behavior and I don't think he'll do any serious damage.

**Comment by daN4cat** [comment]
Agreed, anyway banning is always a bit extreme. Let's live with it.
I removed from wiki pages stuff he added about this ISO image or links to his wiki pages.
I guess there is more and it will take time to fix everything, links included.

**Comment by SteveIreland** [comment]
One of my experiences in life has been to create a message board (pre FaceBook) for the Huntington's Disease community where those that were afflicted with the brain degenerating disease could find some sense of community until they were no longer able to participate.  I found myself continuously editing/deleting posts because those with brain damage have no sense of self-restraint nor context.

I found myself this morning not in a mood to put up with any more posts that trash other members or those needing assistance regardless of the fact that they don't have the ability to read documentation.

If you want me to stop editing his posts then just let me know, otherwise I'll continue to do so.  I've already explained why I did it, so from now on I'll just remove his abusive remarks (which will probably start targeting me now)  :-)

**Comment by SteveIreland** [comment]
I noticed that @daN4cat closed a thread so that only members could add on to the end of it.  I would like to bring an end to users tacking on to the end of a "solidly closed" thread so I thought that would be a great way to start closing all threads.  I know we are sometimes perceived as closing a thread too quickly so if we establish a practice of commenting that a thread is ready to be closed and then giving it a week and if no one adds additional comments then we close it with a restricted lock.  Thoughts?

**Comment by daN4cat** [comment]
I closed a couple of threads to stop the constant trolling, sometimes enough is enough.
Being part of an open source project and giving for free your time doesn't mean that you are in a mission to stand all the human kind of this world in their bad mood and attitudes.
I'm aware sometimes I quickly reply in a dry way, but that's what my time allows me.
And I'm trying to pull the plug a bit because it's becoming too much. Even people that expect everything like being a perfect free product fitting all their needs. I mean, uh?
We worked up to midnight or 1am when we did the UI overall, and these kids expect mummy to even clean after the mess they left behind on the table after their milk and biscuits....

**Comment by daN4cat** [comment]
And no problem if you moderate a bit the discussion or close past posts. It probably makes sense so we avoid random posts.

**Comment by odiea** [comment]
I think I just responded to his rant with something similar only shorter.

**Comment by SteveIreland** [comment]
Yes, and he wants to argue with you so I deleted his post and closed it to further comments.

On another note, the client I "lost" has asked me to reinstate it so I'm highly motivated now to finish up these tax system changes and resume work on the purchasing system and upgrade his system to the latest version.  (Not necessarily in that order).  He told me that the software product that the majority owner paid thousands of dollars for was never finished and that they continue to use OSPOS (even though I haven't given it any love in months).  I just thought you'd appreciate how solid others think OSPOS is (although you already know that).

**Comment by odiea** [comment]
Congrats on your great work. I see he realized what he lost. Both time and money. When we had the thrift store my wife switched over to some bozo who made an inexpensive program. She saw her wrong doing after we had to call him daily with issues or have him reset the program because he did not know how to fix his own issues.

**Comment by daN4cat** [comment]
I always thought that CRTL-ALT-DEL was the solution....

**Comment by daN4cat** [comment]
Glad you got a client back. OSPOS works well, it would just need a smarter sale register with a bit of ajax instead of that boring page refresh, but you can't have everything in your life. I always hoped in a super smart person coming to OSPOS and say: I'll rewrite the sale register and make it fully ajax... still hoping.

**Comment by odiea** [comment]
Unfortunately that never seemed to work with his software. The only way to get out was to do a day close and if you did that you would have to wait until the next day or call him to reset the software.

**Comment by odiea** [comment]
I know us stupid Windows users.

**Comment by daN4cat** [comment]
Guindows

**Comment by jekkos** [comment]
@odiea had a similar experience in fact when at first I didn't think I was able to spend the time building this kind of POS solution and told my mother to find something of the shelve. Same story there, it was full of bugs and just barely usable. Luckily they didn't pay him yet and I told them I'd find something opensource instead, so they let him know they wanted to send all of it in return. He logged into our pc remotely and deleted the whole database that they were filling up since a couple of days. Not the nicest experience.
Luckily I found the phppos back then and so now trying to upgrade them finally to the latest with all these new features. Also I use it to do my own invoicing as a contractor. Couldn't have been nicer.

**Comment by jekkos** [comment]
@daN4cat Maybe we can try to add some bounties for a redesign to attract  a student / frontend dev for this.. would need some wireframes for it first anyway.

**Comment by daN4cat** [comment]
Yep, I'm open to pay for the sale register being ajax with no refresh and open source it after.

**Comment by WebShells** [comment]
@daN4cat @jekkos  I am open as well... 
Agree with you guys, should get away from refreshing thingy in the register... 
For "Guindows" guy, i dont think banning is a good idea too! He can still join with a fake profile and mess around, we cant keep on banning and banning him... Just lets stay away from arguing and opening shitty chats with him...

**Comment by jekkos** [comment]
We might make this public then. I'd be willing to add up a bit of cash as well, but then we should sort of write out the story / technical requirements to make it a bit more concrete on how the architecture should be done. Something responsive would be nice, but it might also mean that the backend needs to be redesigned (most reponses need to be in json then).

**Comment by WebShells** [comment]
@jekkos just noticed that bountysource only supports paypal... In our country, paypal payments aren't supported... Is there any way to add few money?

**Comment by WebShells** [comment]
Guys... You didnt like the below ? 
![ospos logo 2](https://user-images.githubusercontent.com/26513147/45176124-bd04cb00-b217-11e8-8d44-fed317be215c.png)
![ospos logo 3](https://user-images.githubusercontent.com/26513147/45176251-12d97300-b218-11e8-8b7a-ccbd3f274e26.png)
![ospos logo 4](https://user-images.githubusercontent.com/26513147/45176298-38ff1300-b218-11e8-9b74-8979fbf1e8da.png)

**Comment by jekkos** [comment]
Yes I kinda like it, think we should make a decision there, at least it's better than what we have now

**Comment by jekkos** [comment]
You can also pay using crypto but value might decline fast.. no that's fine if you want.

**Comment by jekkos** [comment]
I mean if you donate to opensourcepos paypal I'll add it to the bounty

**Comment by objecttothis** [comment]
@WebShells I prefer the top of those three however, perhaps a fourth variant of the cart and O's being dark and SP and S being lighter blue?

**Comment by objecttothis** [comment]
Sorry for the spam @opensourcepos/org . GitHub is migrating Team Discussions to GitHub Discussions and it spammed everyone. I turned it off for now because GitHub Discussions doesn't have the ability to make discussion categories hidden to all but team or org members. For now let's move those discussions to the gitter dev room. Which curiously also migrated platforms. SMH.

**Comment by Rich-24** [comment]
Hello guys


Please can someone help me on how to add products category in the drop down list so I can just select product category when adding products instant of typing.

Please help🙏🤣

**Reply by objecttothis** [reply]
Go to Settings > General and check this box
![image](https://github.com/user-attachments/assets/5ff1d2a8-5919-4294-ba3b-664089f0f86c)

Go to Attributes and edit the ospos_category attribute
![image](https://github.com/user-attachments/assets/33376405-2654-4695-9f65-5572a38c4f50)

Add your possible dropdown values.
![image](https://github.com/user-attachments/assets/dda20926-891a-455d-a5ac-b3b3135819bf)

Now when an item is created, the user will need to select the category from a dropdown rather than typing it in.

**Reply by Rich-24** [reply]
Wow. You solve what I have been facing as a challenge for a while.
From my heart God richly bless you.

Moreover, 
Please I want to add payment types to what is already in the system.

I don’t have idea about coding. So please if you can help me step by step as you did previously.

Thank you in advance!

**Reply by objecttothis** [reply]
Are you talking about changing these?
![image](https://github.com/user-attachments/assets/ae137efc-ac01-41ce-ad56-9592360dd5bd)

That would require code changes, I believe. Someone could add a feature to be able to add payment methods, but it would take some work.  If people want to make 3rd party integrations to accept online payment methods, this would be a needed change, but the basic feature to have different payment types added wouldn't be too difficult. Submit a feature request if this is something you are wanting.

**Reply by Rich-24** [reply]
Thank you for your response.

But not the online payment types.

I am talking of the payment types at the sales screen where we have ‘cash, credit, debit’ and others. The common one in my country apart from ‘Cash’ is not there. So I wanted to know if you can help me Edit or Add.

**Comment by A7medAmine** [comment]
a yoooo guys can someone explain the logic behind the sale operations and calling the item based on the barcode since i wanna make ospos items able to get unlimited amount of item number in case there are similar products but with different barcodes i've made the table structure and the view and the form for it but i dont understand the logic that i must add to make it work

**Comment by Rich-24** [comment]
Hello guys

Please is it possible to edit this. If yes! Please where can I locate the file

![IMG_1225](https://github.com/user-attachments/assets/9cf38ab7-fce7-46c3-bb6f-79e68d102875)

**Reply by A7medAmine** [reply]
yes absolutely,  by going into  application/language/your language  or EN Us /common_lang.php
edit the value of the common_welcome_message

**Comment by Rich-24** [comment]
Thank you so much

**Comment by Rich-24** [comment]
Hello AHMDabdmine

Please I also search for “OSPOS” in the home. Please in which file can I locate this

![IMG_1226](https://github.com/user-attachments/assets/81d75975-2b91-41a2-954c-1b7e4ab8267e)

**Reply by A7medAmine** [reply]
that's also language i think or on the config file app name

**Reply by A7medAmine** [reply]
you can dm me on Instagram or Facebook

**Comment by Rich-24** [comment]
Hello guys!
I am still looking for the file where I can edit the OSPOS on the HOME screen. Can someone help out?

![IMG_1300](https://github.com/user-attachments/assets/697a7afe-9f2e-416a-9a61-9063a37e70a1)

**Reply by jekkos** [reply]
Did you try company logo upload?

**Reply by A7medAmine** [reply]
you will find it on application/vews/home/home.php I guess

**Reply by A7medAmine** [reply]
ooh I see its on the application/partial/header you will find the ospos toggle

**Comment by Rich-24** [comment]
Yes
It won’t change it

**Comment by Rich-24** [comment]
It isn’t there

**Comment by Rich-24** [comment]
Yeah
I find it now. Thank you very much. Can you share you social media handles
with me.
***@***.***

**Reply by A7medAmine** [reply]
ok  insta 
https://www.instagram.com/ahmeed_amiinee?igsh=ZXc5cW5jdmNjMHoz
email ahmedabdesselam55@gmail.com

**Comment by Rich-24** [comment]
Okay

**Comment by Rich-24** [comment]
Please where can I locate the file to edit the ‘Wholesale Price’ and Retail Price’ name

![IMG_1329](https://github.com/user-attachments/assets/22985835-b6d3-4579-bf9b-a6dd7875c118)

**Comment by Rich-24** [comment]
And please my printer setup options are not enabled. Any settings?
![IMG_1327](https://github.com/user-attachments/assets/581b2584-463d-4b65-abb0-e7a7d53f0e98)

**Reply by A7medAmine** [reply]
it only on old setups with the jsprint fire fox extention installed
